paymentsheet/src/main/java/com/stripe/android/paymentsheet/repositories/StripeCustomerAdapter.kt

```diff
@@ -94,7 +95,11 @@ internal class StripeCustomerAdapter @Inject constructor(
     }
 
     override suspend fun setupIntentClientSecretForCustomerAttach(): Result<String> {
-        TODO()
+        return getCustomerEphemeralKey().mapCatching { customerEphemeralKey ->
+            setupIntentClientSecretProvider?.provide(customerEphemeralKey.customerId)
+        }.getOrElse {
+            Result.failure(it)
+        } ?: throw IllegalArgumentException("setupIntentClientSecretProvider cannot be null")
```

Comment:
I don't hate throwing here. But I think it's also easy to return a result.failure too 🤔

Reply:
I was thinking that too... not sure where we should throw, but I do think throwing somewhere is the right idea. I will do it here for now.